### ISSUE TITLE
Fix disabled warnings in GenerateAnalyzerNuspec

### ISSUE THREAD

**Issue by omajid** [dotnet/roslyn-analyzers#4441]
https://github.com/dotnet/roslyn-analyzers/pull/4436 disables a bunch of warnings in GenerateAnalyzerNuspec:

```csharp
#pragma warning disable CA1820, CS8600, IDE0055, IDE0057, IDE0062, IDE0078, IDE0073
```

We should enable these warnings and fix the code.
